src/main/java/io/jenkins/plugins/checks/github/GitHubChecksContext.java

```diff
@@ -108,4 +111,37 @@ protected boolean hasValidCredentials(final FilteredLog logger) {
     private Optional<GitHubAppCredentials> findGitHubAppCredentials(final String credentialsId) {
         return getScmFacade().findGitHubAppCredentials(getJob(), credentialsId);
     }
+
+    Optional<Long> getId(final String name) {
+        return getAction(name).map(GitHubChecksAction::getId);
+    }
+
+    // New API method
+    ChecksConclusion getConclusion(final String name) {
+        return getAction(name).map(GitHubChecksAction::getConclusion).orElse(ChecksConclusion.NONE);
+    }
+
+    // Implementation using invisible actions
+    private Optional<GitHubChecksAction> getAction(final String name) {
```

Comment:
On the whole, I think I much prefer this way of doing things.

Reply:
agree on this one